components/prize_pool/commons/prize_pool_import.lua

```diff
@@ -0,0 +1,512 @@
+	end
+
+	return Table.deepMergeInto(Import._entryToOpponent(lpdbEntry, placement), Import._removeTbd(entry))
+end
+
+function Import._removeTbd(entry)
+	-- if we get here we know entry is a TBD Opponent
+	-- but it still might hold data (e.g. flag or race of players) for non party opponents
+	local opponent = entry.opponentData
+	if not Opponent.typeIsParty(opponent.type) then
+		-- TBD opponents of non party types do not hold additional data
+		entry.opponentData = {}
+	else
+		opponent.type = nil
+		for _, player in pairs(opponent.players or {}) do
+			if Opponent.playerIsTbd(player) then
+				player.displayName = nil
+				player.pageIsResolved = nil
+				player.pageName = nil
+			end
+		end
+	end
+
+	return entry
+end
+
+function Import._entryToOpponent(lpdbEntry, placement)
+	local additionalData = {}
+	if lpdbEntry.needsLastVs then
+		additionalData = Import._groupLastVsAdditionalData(lpdbEntry)
+	end
+
+	local score = additionalData.score or Import._getScore(lpdbEntry.opponent)
+	local vsScore = additionalData.vsScore or Import._getScore(lpdbEntry.vsOpponent)
+	local lastVsScore
+	if score or vsScore then
+		lastVsScore = (score or '') .. '-' .. (vsScore or '')
+	end
+
+	return placement:_parseOpponents{{
+		Import._removeIfTbd(lpdbEntry.opponent),
+		wdl = (not lpdbEntry.needsLastVs) and Import._makeGroupScore(lpdbEntry) or nil,
+		lastvs = additionalData.lastVs or Import._removeIfTbd(lpdbEntry.vsOpponent),
+		lastvsscore = additionalData.score or lastVsScore,
+		date = lpdbEntry.date,
+		isAlreadyParsed = true,
+	}}[1]
+end
+
+function Import._removeIfTbd(opponent)
+	return (Table.isEmpty(opponent) or Opponent.isTbd(opponent))
+		and {} or Table.merge(opponent, {isAlreadyParsed = true})
+end
+
+function Import._makeGroupScore(lpdbEntry)
+	if not lpdbEntry.scoreBoard then
+		return
+	end
+
+	local matches = lpdbEntry.scoreBoard.match
+	local overtime = lpdbEntry.scoreBoard.overtime
+	local wdl
+	if (matches.d or 0) ~= 0 then
+		wdl = {matches.w or '', matches.d or '', matches.l or ''}
+	elseif (overtime.w or 0) ~= 0 or (overtime.l or 0) ~= 0 then
+		wdl = {matches.w or '', overtime.w or '', overtime.l or '', matches.l or ''}
+	else
+		wdl = {matches.w or '', matches.l or ''}
+	end
+
+	return table.concat(wdl, Import.config.groupScoreDelimiter)
+end
+
+function Import._getScore(opponentData)
+	if not opponentData then
+		return
+	end
+
+	return opponentData.status == SCORE_STATUS and opponentData.score
+		or opponentData.status
+end
+
+function Import._groupLastVsAdditionalData(lpdbEntry)
+	local opponentName = Opponent.toName(lpdbEntry.opponent)
+	local matchConditions = {}
+	for _, matchId in pairs(lpdbEntry.matches) do
+		table.insert(matchConditions, '[[match2id::' .. matchId .. ']]')
+	end
+
+	local matchData = mw.ext.LiquipediaDB.lpdb('match2', {
+		conditions = '[[opponent::' .. opponentName .. ']] AND (' .. table.concat(matchConditions, ' OR ') .. ')',
+		order = 'date desc',
+		query = 'match2opponents, winner',
+		limit = 1
+	})
```

Comment:
Does #527/#528 fix need to be applied? (unless it was fixed while the fixing match2 limit/offset)

Reply:
we could add it or we just hope it was fixed in lpdb

imo we should try it without the additional call to the workaround module
if we encounter issues we can add the call easily then